driver-core/src/test/unit/com/mongodb/connection/ClusterSettingsSpecification.groovy

```diff
@@ -192,6 +205,20 @@ class ClusterSettingsSpecification extends Specification {
         settings.srvMaxHosts == null
         settings.srvServiceName == 'mongodb'
 
+        when:
+        settings = ClusterSettings.builder()
+                .applyConnectionString(new ConnectionString('mongodb://example.com:27018'))
+                .requiredReplicaSetName("test")
+                .build()
+
+        then:
+        settings.mode == ClusterConnectionMode.MULTIPLE
```

Comment:
A note for us in the future: when run against the original code, this assertion fails